InternetHospital.BusinessLogic/Services/ModeratorService.cs

```diff
@@ -0,0 +1,221 @@
+                        dest.SignUpTime = DateTime.UtcNow;
+                        dest.UserName = src.Email;
+                    }));
+                if (await _roleManager.FindByNameAsync(MODERATOR) == null)
+                {
+                    await _roleManager.CreateAsync(new IdentityRole<int>(MODERATOR));
+                }
+                var result = await _userManager.CreateAsync(user, vm.Password);
+                if (result.Succeeded)
+                {
+                    await _userManager.AddToRoleAsync(user, MODERATOR);
+                    return (user, "Confirm link was send on email");
+                }
+
+                return (null, "Error during registration");
+            }
+            return (null, "User with such email already exists!");
+        }
+
+        public FilteredModeratorsModel GetFilteredModerators(ModeratorSearchParameters queryParameters)
+        {
+            var moderators = GetModerators();
+
+            if (!queryParameters.IncludeAll)
+            {
+                moderators = moderators.Where(m => m.Status.Name == ACTIVE_STATUS);
+            }
+
+            if (!string.IsNullOrEmpty(queryParameters.SearchByName))
+            {
+                var lowerSearchParam = queryParameters.SearchByName.ToLower();
+                moderators = moderators.Where(m => m.FirstName.ToLower().Contains(lowerSearchParam)
+                                                   || m.SecondName.ToLower().Contains(lowerSearchParam)
+                                                   || m.ThirdName.ToLower().Contains(lowerSearchParam));
+            }
+
+            if (!string.IsNullOrEmpty(queryParameters.Order))
+            {
+                moderators = SortModerators(moderators, queryParameters.Sort, queryParameters.Order);
+            }
+
+            FilteredModeratorsModel fModel = new FilteredModeratorsModel();
+            fModel.AmountOfAllFiltered = moderators.Count();
+
+            moderators = PaginationHelper<User>
+                .GetPageValues(moderators, queryParameters.PageSize, queryParameters.Page);
+
+            fModel.Moderators = ConvertToViewModel(moderators).ToList();
+            return fModel;
+        }
+
+        public async Task<(bool,string)> DeleteAsync(int id)
+        {
+            var user = await _userManager.FindByIdAsync(id.ToString());
+            if (user != null && await _userManager.IsInRoleAsync(user, "Moderator"))
+            {
+                await _userManager.UpdateAsync(user);
+                user.StatusId = _context.Statuses.FirstOrDefault(s => s.Name == DELETED_STATUS).Id;
+                await _context.SaveChangesAsync();
+                return (true, "Deleted successfully");
+            }
+
+            return (false, "Couldn't delete user with such id!");
+        }
+
+        public async Task<IEnumerable<(bool, string)>> DeleteAsync(int[] ids)
+        {
+            List<(bool, string)> messages = new List<(bool, string)>();
```

Comment:
I don't think that it's a great idea to create List<T> with tuples inside and return him. You will get only one result in this method.

Reply:
It's for giving possibility to process the result of every delete.